src/utils/config.jl

```diff
@@ -0,0 +1,258 @@
+# TODO (later): move this definition to external files
+global DEFAULT_CONFIG::Dict{String,Any} = Dict{String,Any}(
+    "performance" => Dict{String,Any}(
+        "full_analysis" => Dict{String,Any}(
+            "debounce" => 1.0,
+            "throttle" => 5.0
+        )
+    ),
+    "testrunner" => Dict{String,Any}(
+        "executable" => "testrunner"
+    ),
+)
+
+global CONFIG_RELOAD_REQUIRED::Dict{String,Any} = Dict{String,Any}(
+    "performance" => Dict{String,Any}(
+        "full_analysis" => Dict{String,Any}(
+            "debounce" => true,
+            "throttle" => true
+        )
+    ),
+    "testrunner" => Dict{String,Any}(
+        "executable" => false
+    ),
+)
+
+function access_nested_dict(dict::Dict{String,Any}, path::String, rest_path::String...)
+    nextobj = @something get(dict, path, nothing) return nothing
+    if !(nextobj isa Dict{String,Any})
+        if isempty(rest_path)
+            return nextobj
+        else
+            return nothing
+        end
+    end
+    return access_nested_dict(nextobj, rest_path...)
+end
+
+"""
+    traverse_merge!(on_leaf::Function, target::Dict{String,Any},
+                    source::Dict{String,Any}, key_path::Vector{String})
+
+Recursively merges `source` into `target`, traversing nested dictionaries.
+If a key in `source` is a dictionary, it will recursively merge it into the corresponding key in `target`.
+If a key in `source` is not a dictionary, the `on_leaf` function is called with:
+- `target`: the target dictionary being modified
+- `k`: the key from `source`
+- `v`: the value from `source`
+- `path`: the current path as a vector of strings
+
+If the key does not exist in `target`, it will be created as an empty dictionary before merging.
+"""
+function traverse_merge!(on_leaf::Function, target::Dict{String,Any},
+                         source::Dict{String,Any}, key_path::Vector{String})
+    for (k, v) in source
+        current_path = [key_path; k]
+        if v isa Dict{String,Any}
+            tv = get(target, k, nothing)
+            if tv isa Dict{String,Any}
+                traverse_merge!(on_leaf, tv, v, current_path)
+            else
+                tv = target[k] = Dict{String,Any}()
+                traverse_merge!(on_leaf, tv, v, current_path)
+            end
```

Comment:
```suggestion
            tv = get!(Dict{String,Any}, target, k)
            traverse_merge!(on_leaf, tv, v, current_path)
```

Reply:
There are cases where `tv` is not a `Dict` (like `String`), and even in those cases, we still need to create an empty dict. So I think the current implementation is the most straightforward.